src/components/AddressResolver/EndpointEntry.tsx

```diff
@@ -0,0 +1,164 @@
+import React, { useState } from "react";
+import styled from "@emotion/styled";
+import { InputDefault } from "../UI/Input";
+import { SvgIcon, SvgIconTrash2, SvgIconSave, SvgIconEdit2 } from "../UI/SvgIcon";
+import { useThirdPartyAPIEndpoints } from "../../common/hooks/useThirdPartyAPIEndpoints";
+import { vars } from "../../styles";
+import { generateUniqueId } from "./../../common/utils/generateUniqueId";
+import isURL from "validator/lib/isURL";
+import isEmpty from "validator/lib/isEmpty";
+
+interface EndpointEntryProps {
+  className?: string;
+  id: string;
+  order: number;
+  removeEndpoint: () => void;
+  api?: string;
+  name?: string;
+  canEdit?: boolean;
+}
+
+export const EndpointEntry = styled(
+  ({ className, id, order, removeEndpoint, api = "", name = "", canEdit = false }: EndpointEntryProps) => {
```

Comment:
This is quite strange, if you are putting default value, your props dont have to have `?` i think. But i dont understand why you need the default values in the first place.